tests/MongoDB.Driver.Tests/Specifications/crud/CrudOperationTestBase.cs

```diff
@@ -44,8 +48,24 @@ public void Execute(ClusterDescription clusterDescription, IMongoDatabase databa
                 }
             }
 
-            Execute(collection, outcome, async);
+            try
+            {
+                Execute(collection, outcome, async);
+            }
+            catch (Exception ex)
+            {
+                ActualError = ex;
```

Comment:
Saving the Actual error and hiding the exception in case if the expected test result is an exception.
NOTE: Currently the `ActualException` we used only as a flag about whether any exception has been thrown or not, so we can also save only `bool` flag instead of a full error object, but we may need the full exception in future.